common/ledger/snapshot/file.go

```diff
@@ -0,0 +1,200 @@
+		multiWriter:       multiWriter,
+		hasher:            hasher,
+		varintReusableBuf: make([]byte, binary.MaxVarintLen64),
+	}, nil
+}
+
+// EncodeString encodes and appends the string to the data stream
+func (c *FileCreator) EncodeString(str string) error {
+	return c.EncodeBytes([]byte(str))
+}
+
+// EncodeString encodes and appends a proto message to the data stream
+func (c *FileCreator) EncodeProtoMessage(m proto.Message) error {
+	b, err := proto.Marshal(m)
+	if err != nil {
+		return errors.Wrapf(err, "error marshalling proto message to write to the snapshot file: %s", c.file.Name())
+	}
+	return c.EncodeBytes(b)
+}
+
+// EncodeBytes encodes and appends bytes to the data stream
+func (c *FileCreator) EncodeBytes(b []byte) error {
+	if err := c.EncodeUVarint(uint64(len(b))); err != nil {
+		return err
+	}
+	if _, err := c.multiWriter.Write(b); err != nil {
+		return errors.Wrapf(err, "error while writing data to the snapshot file: %s", c.file.Name())
+	}
+	return nil
+}
+
+// EncodeUVarint encodes and appends a number to the data stream
+func (c *FileCreator) EncodeUVarint(u uint64) error {
+	n := binary.PutUvarint(c.varintReusableBuf, u)
+	if _, err := c.multiWriter.Write(c.varintReusableBuf[:n]); err != nil {
+		return errors.Wrapf(err, "error while writing data to the snapshot file: %s", c.file.Name())
+	}
+	return nil
+}
+
+// Done closes the snapshot file and returns the final hash of the data stream
+func (c *FileCreator) Done() ([]byte, error) {
+	if err := c.bufWriter.Flush(); err != nil {
+		return nil, errors.Wrapf(err, "error while flushing to the snapshot file: %s ", c.file.Name())
+	}
+	if err := c.file.Close(); err != nil {
+		return nil, errors.Wrapf(err, "error while closing the snapshot file: %s ", c.file.Name())
+	}
+	return c.hasher.Sum(nil), nil
+}
+
+// Close closes the underlying file, if not already done. A consumer can invoke this function if the consumer
+// encountered some error and simply wants to abandon the snapshot file creation (typically, intended to be used in a defer statement)
+func (c *FileCreator) Close() error {
+	if c == nil {
+		return nil
+	}
+	return errors.Wrapf(c.file.Close(), "error while closing the snapshot file: %s", c.file.Name())
+}
+
+// FileReader reads from a ledger snapshot file. This is expected to be used for loading the ledger snapshot data
+// during bootstrapping a channel from snapshot. The data should be read, using the functions `DecodeXXX`,
+// in the same sequence in which the data was written by the functions `EncodeXXX` in the `FileCreator`.
+// Note that the FileReader does not verifies the hash of stream and it is expected that the hash has been verified
+// by the consumer. Later, if we decide to perform this, on-the-side, while loading the snapshot data, the FileRedear,
+// like the FileCreator, would take a `hasher` as an input
+type FileReader struct {
+	file              *os.File
+	bufReader         *bufio.Reader
+	reusableByteSlice []byte
```

Comment:
```suggestion
	varintReusableBuf []byte
```
to maintain consistency with what we have defined in the `FileCreator`

Reply:
Both are different. unlike varintReusableBuf, this buffer is for random bytes of arbitrary length